.github/workflows/conda-lock.yml

```diff
@@ -32,7 +32,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: ${{ github.event.client_payload.pull_request.head.repo.full_name }}
-          ref: ${{ github.event.client_payload.pull_request.head.ref }}
+          ref: ${{ github.ref_name }}
```

Comment:
Hopefully this will allow the conda-lock GitHub Action workflow to push to the Pull Request branch (instead of the main branch as mentioned at https://github.com/weiji14/conda-lock-refresh-demo/pull/2#issuecomment-1689122615). Will need to merge this into the main branch to test properly.

Reply:
Actually, should probably use `refs/pull/${{ github.event.issue.number }}/head` instead, following a few links from https://github.com/orgs/community/discussions/49628#discussioncomment-5478453. Patched directly on main branch at 1631a18f92d1a7188cab73d9e02ae0076dd23987

Reply:
Third time lucky hopefully. Should use `hub pr checkout ${{ github.event.issue.number }}` to checkout the git branch instead of having a detached HEAD, see patch 0f280e78657578326ba74d9f13861235661fb2d9.